### PR TITLE
Refine compositor API

### PR DESCRIPTION
**Pull request by vberger**
Remove unneeded liveness checks and introduce pro/post commit hooks as well as destruction hooks for surfaces.

### CODE REVIEW

File: src/wayland/compositor/mod.rs [hunk 1]
```diff
@@ -402,18 +381,26 @@ pub fn get_region_attributes(region: &wl_region::WlRegion) -> RegionAttributes {
...
-    }
-    PrivateSurfaceData::add_commit_hook(surface, hook)
+/// It'll be invoked on surface commit, *before* the new state is merged into the current state.
+pub fn add_pre_commit_hook(surface: &WlSurface, hook: fn(&mut DisplayHandle<'_>, &WlSurface)) {
+    PrivateSurfaceData::add_pre_commit_hook(surface, hook)
+}
+
+/// Register a post-commit hook to be invoked on surface commit
+///
+/// It'll be invoked on surface commit, *before* the new state is merged into the current state.
```
Comment (Drakulix) [inline src/wayland/compositor/mod.rs:393]:
```suggestion
/// It'll be invoked on surface commit, *after* the new state is merged into the current state.
```

File: src/wayland/compositor/mod.rs [hunk 2]
```diff
@@ -402,18 +381,26 @@ pub fn get_region_attributes(region: &wl_region::WlRegion) -> RegionAttributes {
...
+pub fn add_post_commit_hook(surface: &WlSurface, hook: fn(&mut DisplayHandle<'_>, &WlSurface)) {
+    PrivateSurfaceData::add_post_commit_hook(surface, hook)
+}
+
+/// Register a pre-commit hook to be invoked on surface commit
+///
+/// It'll be invoked when the surface is destroyed (either explicitly by the client or on
+/// client disconnect).
```
Comment (Drakulix) [inline src/wayland/compositor/mod.rs:401]:
```suggestion
/// Register a destruction hook to be invoked on surface destruction.
///
/// It'll be invoked when the surface is destroyed (either explicitly by the client or on
/// client disconnect).
```

**Review by Drakulix — COMMENTED** [review]
Do we want to make use and test the new hooks in this PR as well or leave that for future PRs?

**Comment by vberger** [comment]
> Do we want to make use and test the new hooks in this PR as well or leave that for future PRs?

I'd suggest in later work, given the 0.30 update is still quite WIP I'm not sure everything else is in place to use it in the rederer code yet.

**Pull request merged**